wallet/src/main/java/bisq/wallet/GrpcWalletService.java

```diff
@@ -0,0 +1,248 @@
+                });
+    }
+
+    @Override
+    public Observable<Coin> getBalance() {
+        return balance;
+    }
+
+    @Override
+    public ObservableSet<Transaction> getTransactions() {
+        return transactions;
+    }
+
+    @Override
+    public CompletableFuture<ObservableSet<Transaction>> requestTransactions() {
+        ListTransactionsRequest request = ListTransactionsRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.listTransactions(request))
+                .thenApply(response -> {
+                    transactions.clear();
+                    transactions.addAll(response.getTransactionsList().stream()
+                            .map(this::toTransaction)
+                            .toList());
+                    return transactions;
+                });
+    }
+
+    @Override
+    public CompletableFuture<Boolean> shutdown() {
+        channel.shutdown();
+        return CompletableFuture.completedFuture(true);
+    }
+
+    private Transaction toTransaction(bisq.wallets.grpc.pb.Transaction tx) {
+        return new Transaction(
+                tx.getTxId(),
+                tx.getInputsList().stream().map(this::toTransactionInput).toList(),
+                tx.getOutputsList().stream().map(this::toTransactionOutput).toList(),
+                tx.getLockTime(),
+                tx.getHeight(),
+                Optional.of(new Date(tx.getDate() * 1000)),
+                tx.getConfirmations(),
+                tx.getAmount(),
+                tx.getIncoming()
+        );
+    }
+
+    private bisq.wallet.vo.TransactionInput toTransactionInput(bisq.wallets.grpc.pb.TransactionInput input) {
+        return new bisq.wallet.vo.TransactionInput(
+                input.getPrevOutTxId(),
+                input.getPrevOutIndex(),
+                input.getSequence(),
+                input.getScriptSig(),
+                input.getWitness()
+        );
+    }
+
+    private bisq.wallet.vo.TransactionOutput toTransactionOutput(bisq.wallets.grpc.pb.TransactionOutput output) {
+        return new bisq.wallet.vo.TransactionOutput(
+                output.getValue(),
+                output.getAddress(),
+                output.getScriptPubKey()
+        );
+    }
+
+    private static <T> CompletableFuture<T> toCompletableFuture(ListenableFuture<T> future) {
+        CompletableFuture<T> completableFuture = new CompletableFuture<>();
+        Futures.addCallback(future, new FutureCallback<T>() {
+            @Override
+            public void onSuccess(T result) {
+                completableFuture.complete(result);
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                completableFuture.completeExceptionally(t);
+            }
+        }, Executors.newSingleThreadExecutor());
+        return completableFuture;
+    }
+
+    private record GrpcTransactionInfo(bisq.wallets.grpc.pb.Transaction tx) implements TransactionInfo {
```

Comment:
Move out into vo package. Dont prefix with Grpc remove interface and use that impl directly.

Reply:
Actually I dont see the reason for that class. Its just a subset of tx. I guess we can remove it and use Tx instead